Seems to fix it. While we're here, maybe we should clean up the Shellcheck warnings, and note why we should ignore the return value in a comment:

```
$ shellcheck --shell=bash build-support/setup-hooks/auto-patchelf.sh

In ../../../../build-support/setup-hooks/auto-patchelf.sh line 14:
addEnvHooks "$targetOffset" gatherLibraries
             ^-----------^ SC2154: targetOffset is referenced but not assigned.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 119:
        if [ -n "$runtimeDependencies" ]; then
                 ^------------------^ SC2154: runtimeDependencies is referenced but not assigned.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 132:
    local missing="$(
          ^-----^ SC2155: Declare and assign separately to avoid masking return values.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 178:
      $(find "$@" "${findOpts[@]}" \! -type d \
      ^-- SC2044: For loops over find output are fragile. Use find -exec or a while read loop.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 214:
      [ -n "$(echo "$segmentHeaders" | grep '^Program Headers:')" ] || continue
        ^-- SC2143: Use grep -q instead of comparing output with [ -n .. ].


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 217:
          [ -n "$(echo "$segmentHeaders" | grep "^ *INTERP\\>")" ] || continue
            ^-- SC2143: Use grep -q instead of comparing output with [ -n .. ].


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 233:
    if [[ $depsMissing == 1 && -z "$autoPatchelfIgnoreMissingDeps" ]]; then
                                   ^----------------------------^ SC2154: autoPatchelfIgnoreMissingDeps is referenced but not assigned.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 247:
postFixupHooks+=('
                 ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.

For more information:
  https://www.shellcheck.net/wiki/SC2044 -- For loops over find output are fr...
  https://www.shellcheck.net/wiki/SC2154 -- autoPatchelfIgnoreMissingDeps is ...
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
```

--- a/pkgs/build-support/setup-hooks/auto-patchelf.sh
+++ b/pkgs/build-support/setup-hooks/auto-patchelf.sh
@@ -1,3 +1,5 @@
+#!/usr/bin/env bash
+
 declare -a autoPatchelfLibs
 declare -Ag autoPatchelfFailedDeps
 
@@ -11,6 +13,8 @@ runPatchelf() {
   patchelf "$@" || (echo "Command failed: patchelf $*" && exit 1)
 }
 
+# shellcheck disable=SC2154
+# (targetOffset is referenced but not assigned.)
 addEnvHooks "$targetOffset" gatherLibraries
 
 isExecutable() {
@@ -116,6 +120,8 @@ autoPatchelfFile() {
     interpreter="$(< "$NIX_CC/nix-support/dynamic-linker")"
     if isExecutable "$toPatch"; then
         runPatchelf --set-interpreter "$interpreter" "$toPatch"
+        # shellcheck disable=SC2154
+        # (runtimeDependencies is referenced but not assigned.)
         if [ -n "$runtimeDependencies" ]; then
             for dep in $runtimeDependencies; do
                 rpath="$rpath${rpath:+:}$dep/lib"
@@ -129,10 +135,11 @@ autoPatchelfFile() {
     # clear the RPATH first.
     runPatchelf --remove-rpath "$toPatch"
 
-    local missing="$(
+    local missing
+    missing="$(
         ldd "$toPatch" 2> /dev/null | \
             sed -n -e 's/^[\t ]*\([^ ]\+\) => not found.*/\1/p'
-    )"
+    )" || return 0
 
     # This ensures that we get the output of all missing dependencies instead
     # of failing at the first one, because it's more useful when working on a
@@ -174,10 +181,10 @@ addAutoPatchelfSearchPath() {
         esac
     done
 
-    for file in \
-      $(find "$@" "${findOpts[@]}" \! -type d \
-          \( -name '*.so' -o -name '*.so.*' \))
-    do addToDepCache "$file"; done
+    while IFS= read -r -d '' file; do
+    addToDepCache "$file"
+    done <  <(find "$@" "${findOpts[@]}" \! -type d \
+            \( -name '*.so' -o -name '*.so.*' \) -print0)
 }
 
 autoPatchelf() {
@@ -230,6 +237,8 @@ autoPatchelf() {
       echo "autoPatchelfHook could not satisfy dependency $failedDep wanted by ${autoPatchelfFailedDeps[$failedDep]}"
       depsMissing=1
     done
+    # shellcheck disable=SC2154
+    # (autoPatchelfIgnoreMissingDeps is referenced but not assigned.)
     if [[ $depsMissing == 1 && -z "$autoPatchelfIgnoreMissingDeps" ]]; then
       echo "Add the missing dependencies to the build inputs or set autoPatchelfIgnoreMissingDeps=true"
       exit 1

--- a/pkgs/build-support/setup-hooks/auto-patchelf.sh
+++ b/pkgs/build-support/setup-hooks/auto-patchelf.sh
@@ -135,6 +135,8 @@ autoPatchelfFile() {
     # clear the RPATH first.
     runPatchelf --remove-rpath "$toPatch"
 
+    # If the file is not a dynamic executable, ldd/sed will fail,
+    # in which case we return, since there is nothing left to do.
     local missing
     missing="$(
         ldd "$toPatch" 2> /dev/null | \

